client/landing/stepper/declarative-flow/internals/steps-repository/playground/components/playground-setup/index.tsx

```diff
@@ -0,0 +1,82 @@
+import { useI18n } from '@wordpress/react-i18n';
+import { getQueryArg } from '@wordpress/url';
+import { PlaygroundClient } from '@wp-playground/client';
+import { useRef } from 'react';
+import DocumentHead from 'calypso/components/data/document-head';
+import Loading from 'calypso/components/loading';
+import StepWrapper from 'calypso/signup/step-wrapper';
+import { useIsPlaygroundEligible } from '../../../../../../hooks/use-is-playground-eligible';
+import { importPlaygroundSite } from '../../lib/import-playground';
+import { PlaygroundIframe } from '../playground-iframe';
+import type { Step } from '../../../../types';
+import './style.scss';
+
+export const PlaygroundSetupStep: Step< {
+	submits: {
+		siteSlug?: string;
+		siteId: number;
+	};
+} > = ( props ) => {
+	const { submit } = props.navigation;
+	const isPlaygroundEligible = useIsPlaygroundEligible();
+	if ( ! isPlaygroundEligible ) {
+		window.location.assign( '/start' );
+	}
```

Comment:
I would move this to the bottom and do 
```suggestion
	if ( ! isPlaygroundEligible ) {
		return window.location.assign( '/start' );
	}
```

This would prevent the component from rendering for a second before redirecting.